odbc-api/src/handles/connection.rs

```diff
@@ -327,4 +333,141 @@ impl<'c> Connection<'c> {
         .into_result(self)?;
         Ok(out)
     }
+
+    /// Query all columns that match the provided catalog name, schema pattern, table pattern, and
+    /// column pattern.
+    pub fn columns(
+        &self,
+        statement: StatementImpl<'c>,
+        catalog_name: &U16Str,
+        schema_name: &U16Str,
+        table_name: &U16Str,
+        column_name: &U16Str,
+    ) -> Result<Option<CursorImpl<'c, StatementImpl<'c>>>, Error> {
+        columns(
+            statement,
+            catalog_name,
+            schema_name,
+            table_name,
+            column_name,
+        )
+    }
+
+    /// The buffer descriptions for all standard buffers (not including extensions) returned in the
+    /// columns query (e.g. [`Connection::columns`]).
+    pub fn columns_buffer_description(&self) -> Result<Vec<BufferDescription>, Error> {
+        let null_i16 = BufferDescription {
+            kind: BufferKind::I16,
+            nullable: true,
+        };
+
+        let not_null_i16 = BufferDescription {
+            kind: BufferKind::I16,
+            nullable: false,
+        };
+
+        let null_i32 = BufferDescription {
+            kind: BufferKind::I32,
+            nullable: true,
+        };
+
+        // The definitions for these descriptions are taken from the documentation of `SQLColumns`
+        // located at https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolumns-function
+        let catalog_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_catalog_name_len()? as usize,
+            },
+            nullable: true,
+        };
+
+        let schema_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_schema_name_len()? as usize,
+            },
+            nullable: true,
+        };
+
+        let table_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_table_name_len()? as usize,
+            },
+            nullable: false,
+        };
+
+        let column_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_column_name_len()? as usize,
+            },
+            nullable: false,
+        };
+
+        let data_type_desc = not_null_i16;
+
+        const MAX_EXPECTED_TYPE_NAME_LEN: usize = 255;
```

Comment:
Are these constants indigenous to the ODBC standard? If so we can think about pushing them down to `odbc-sys`.

Reply:
I don't think so. I wasn't sure what to do with these so I tried to assume reasonable defaults that are probably big enough to avoid truncation. I see a few options:
- (current behavior) Use constants with some reasonable value. If truncation happens, then the caller will need to create their own fields if they want to avoid truncation.
- Accept these lengths as arguments so we don't have to hard-code a default.
- Use a default and detect truncation after the fact/increase the limit. This would only be possible in a higher level wrapper though, because we can't tell if something has been truncated until we process each record.

Reply:
If ODBC itself doesn't give any guideline here, I'd suggest forwarding these to the caller

Reply:
Sounds good, I'll make that change

Reply:
Done